test/torchaudio_unittest/prototype/functional/functional_test_impl.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+import torch
+import torchaudio.prototype.functional as F
+from scipy import signal
+from torchaudio_unittest.common_utils import nested_params, TestBaseMixin
+
+
+class FunctionalTestImpl(TestBaseMixin):
+    @nested_params(
+        [F.convolve, F.fftconvolve],
+        [(10, 4), (4, 3, 1, 2), (2,), ()],
+        [(100, 43), (21, 45)],
+    )
+    def test_convolve_numerics(self, fn, leading_dims, lengths):
+        """Check that convolve returns values identical to those that SciPy produces."""
+        L_x, L_y = lengths
+
+        x = torch.rand(*(leading_dims + (L_x,)), dtype=self.dtype, device=self.device)
+        y = torch.rand(*(leading_dims + (L_y,)), dtype=self.dtype, device=self.device)
+
+        actual = fn(x, y)
+
+        num_signals = torch.tensor(leading_dims).prod() if leading_dims else 1
+        x_reshaped = x.reshape((num_signals, L_x))
+        y_reshaped = y.reshape((num_signals, L_y))
+        expected = [
+            signal.convolve(x_reshaped[i].detach().cpu().numpy(), y_reshaped[i].detach().cpu().numpy())
```

Comment:
`scipy.signal.fftconvolve` can be applied on N-D arrays by setting `axes` to `[-1]`, so that we can optimize the testing speed.
To compare `F.convolve` with `scipy.signal.convolve` there is no `axes` argument, we have to use list iteration.